### PR TITLE
remove ActorPublisher implementations

### PR DESCRIPTION
**Pull request by bardurdam**
As `streamPublisher` and `allStreamsPublisher` contracts define reactive streams publishers, so we can still uphold the contract by using `streamSource` and a sink that produces a reactive stream publisher.

The methods are deprecated, because the are just for convenience and to signal that they should only be used when integrating with other reactive streams library than Akka Stream.

### CODE REVIEW

File: src/main/scala/eventstore/examples/MessagesPerSecondReactiveStreams.scala [hunk 1]
```diff
@@ -4,16 +4,20 @@ import akka.actor.ActorSystem
...
 import scala.concurrent.duration._
 
 object MessagesPerSecondReactiveStreams extends App {
   implicit val system = ActorSystem()
   implicit val materializer = ActorMaterializer()
   val connection = EventStoreExtension(system).connection
-  val publisher = connection.allStreamsPublisher()
 
-  Source.fromPublisher(publisher)
+  def subscriber: Subscriber[String] = ??? // subscriber defined by another library
```
Comment (t3hnar) [inline src/main/scala/eventstore/examples/MessagesPerSecondReactiveStreams.scala]: please make up some dummy subscriber

File: README.md [hunk 2]
```diff
@@ -412,50 +412,61 @@ EventStoreExtension(system).actor ! ReadEvent(EventStream.Id("stream"))
...
   implicit val system = ActorSystem()
-  import system.dispatcher
   implicit val materializer = ActorMaterializer()
   val connection = EventStoreExtension(system).connection
-  val publisher = connection.streamPublisher(EventStream.System.`$streams`, infinite = false, resolveLinkTos = true)
-  Source.fromPublisher(publisher)
-    .runForeach { x => println(x.streamId.streamId) }
-    .onComplete{ _ => system.terminate()}
+
+  def subscriber: Subscriber[String] = ??? // subscriber defined by another library
```
Comment (t3hnar) [inline README.md]: please make up some dummy subscriber

File: src/main/scala/eventstore/examples/MessagesPerSecondReactiveStreams.scala (hunk 1 above)
Comment (bardurdam) [reply src/main/scala/eventstore/examples/MessagesPerSecondReactiveStreams.scala]: Sure

**Comment by bardurdam** [comment]
@t3hnar I've updated the PR with the changes you requested

**Pull request merged**